troi/patches/lb_radio_classes/artist.py

```diff
@@ -155,24 +139,28 @@ def read(self, entities):
             self.local_storage["user_feedback"].append(msg)
         self.data_cache["element-descriptions"].append("artist %s" % artists[0]["name"])
 
+        artist_mbids = [ artist["mbid"] for artist in artists ]
+        artist_mbids = list(set(artist_mbids))
+        artist_recordings = self.recording_search_by_artist.search(artist_mbids, start, stop, self.max_top_recordings_per_artist)
+
         # Now collect recordings from the artist and similar artists and return an interleaved
-        # strem of recordings.
+        # stream of recordings.
         for i, artist in enumerate(artists):
-            if artist["mbid"] + "_top_recordings" in self.data_cache:
-                artist["recordings"] = self.data_cache[artist["mbid"] + "_top_recordings"]
-                continue
 
-            recs_plist = plist(self.fetch_top_recordings(artist["mbid"]))
+            # TODO: This disables top recordings caching, which needs to be re-thought given the new approach
+            #if artist["mbid"] + "_top_recordings" in self.data_cache:
+            #    artist["recordings"] = self.data_cache[artist["mbid"] + "_top_recordings"]
+            #    continue
+
+            recs_plist = plist(artist_recordings[artist["mbid"]])
             if len(recs_plist) < 20:
                 self.local_storage["user_feedback"].append(
                     f"Artist {artist['name']} only has {'no' if len(recs_plist) == 0 else 'few'} top recordings.")
 
-            recordings = []
-            for recording in recs_plist.random_item(start, stop, self.max_top_recordings_per_artist):
-                recordings.append(recording)
+            recordings = recs_plist.random_item(start, stop, self.max_top_recordings_per_artist)
 
             # Now tuck away the data for caching and interleaving
-            self.data_cache[artist["mbid"] + "_top_recordings"] = recordings
+            #self.data_cache[artist["mbid"] + "_top_recordings"] = recordings
```

Comment:
intentional?

Reply:
Nope, will remove.

Reply:
Left for a future release. The current release goal is to get Funkwhale moving again, but next week I will make another pass over troi and then work on a new release for lb-server.